test/bootc/utils.sh

```diff
@@ -0,0 +1,305 @@
+#! /usr/bin/env bash
+
+# Import util functions
+source "$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)/../rpm/utils.sh"
+
+build_bootc_container() {
+  local base_image=$1
+
+  if [[ ${ID} == "fedora" ]]; then
+    tee Containerfile > /dev/null << EOF
+FROM ${base_image}
+RUN cat > /etc/yum.repos.d/fedora-iot.repo << 'REPO_EOF'
+[copr:copr.fedorainfracloud.org:group_fedora-iot:fedora-iot]
+name=Copr repo for fedora-iot owned by @fedora-iot
+baseurl=https://download.copr.fedorainfracloud.org/results/@fedora-iot/fedora-iot/fedora-\$releasever-\$basearch/
+type=rpm-md
+skip_if_unavailable=True
+gpgcheck=1
+gpgkey=https://download.copr.fedorainfracloud.org/results/@fedora-iot/fedora-iot/pubkey.gpg
+repo_gpgcheck=0
+enabled=1
+enabled_metadata=1
+REPO_EOF
+RUN dnf install -y go-fdo-client
+EOF
+  elif [[ ${ID} == "centos" ]]; then
+    tee Containerfile > /dev/null << EOF
+FROM ${base_image}
+RUN cat > /etc/yum.repos.d/fedora-iot.repo << 'REPO_EOF'
+[copr:copr.fedorainfracloud.org:group_fedora-iot:fedora-iot]
+name=Copr repo for fedora-iot owned by @fedora-iot
+baseurl=https://download.copr.fedorainfracloud.org/results/@fedora-iot/fedora-iot/centos-stream-\$releasever-\$basearch/
+type=rpm-md
+skip_if_unavailable=True
+gpgcheck=1
+gpgkey=https://download.copr.fedorainfracloud.org/results/@fedora-iot/fedora-iot/pubkey.gpg
+repo_gpgcheck=0
+enabled=1
+enabled_metadata=1
+REPO_EOF
+RUN dnf install -y go-fdo-client
+EOF
+  fi
+
+  podman build --retry=5 --retry-delay=10s -t "fdo-bootc:latest" -f Containerfile .
+}
```

Comment:
Do not generate the copr repos manually, the above can be simplified by using:
```bash
rpm -q --whatprovides 'dnf-command(copr)' &> /dev/null || dnf install -y 'dnf-command(copr)'; 
dnf copr enable -y '@fedora-iot/fedora-iot'
dnf install -y go-fdo-client
dnf copr disable -y @fedora-iot/fedora-iot
```

Reply:
Actually is a bit trickier:
```Dockerfile
RUN dnf=\$(readlink \$(command -v dnf)); [ "\${dnf}" = "dnf5" ] || dnf=dnf ; \
    rpm -q --whatprovides \${dnf}'-command(copr)' &> /dev/null || \${dnf} install -y \${dnf}'-command(copr)'; \
    \${dnf} copr enable -y '@fedora-iot/fedora-iot'; \
    \${dnf} install -y go-fdo-client; \
    \${dnf} copr disable -y @fedora-iot/fedora-iot
EOF
```

The dnf command for dnf4 is `dnf-command(copr)` but for dnf5 is `dnf5-command(copr)`

Reply:
Done